x-pack/plugins/aiops/public/components/change_point_detection/charts_grid.tsx

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { type FC, useMemo, useState } from 'react';
+import {
+  EuiBadge,
+  EuiDescriptionList,
+  EuiFlexGrid,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiHorizontalRule,
+  EuiIcon,
+  EuiPagination,
+  EuiPanel,
+  EuiText,
+  EuiToolTip,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { SelectedChangePoint } from './change_point_detection_context';
+import { ChartComponent } from './chart_component';
+
+const CHARTS_PER_PAGE = 6;
+
+interface ChartsGridProps {
+  changePoints: Record<number, SelectedChangePoint[]>;
+}
+
+export const ChartsGrid: FC<ChartsGridProps> = ({ changePoints: changePointsDict }) => {
+  const changePoints = useMemo(() => {
+    return Object.values(changePointsDict).flat();
+  }, [changePointsDict]);
+
+  const [activePage, setActivePage] = useState<number>(0);
+
+  const resultPerPage = useMemo(() => {
+    const start = activePage * CHARTS_PER_PAGE;
+    return changePoints.slice(start, start + CHARTS_PER_PAGE);
+  }, [changePoints, activePage]);
+
+  const pagination = useMemo(() => {
+    return {
+      activePage,
+      pageCount: Math.round((changePoints.length ?? 0) / CHARTS_PER_PAGE),
+      updatePagination: setActivePage,
+    };
+  }, [activePage, changePoints.length]);
+
+  return (
+    <>
+      <EuiFlexGrid columns={resultPerPage.length >= 2 ? 2 : 1} responsive gutterSize={'m'}>
+        {resultPerPage.map((v) => {
+          return (
+            <EuiFlexItem key={v.group?.value ?? 'single_metric'}>
```

Comment:
With the update in this PR to be able to select items from different groups, I was able to hit a `Warning: Encountered two children with the same key` for this. For example, with the `cloudwatch` dataset I create this setup (NetworkIn/NetworkOut by region):

<img width="1656" alt="image" src="https://user-images.githubusercontent.com/230104/231422311-93a70be5-bb91-4b5f-8857-98829a1d9d73.png">

Then from both metrics I selected the same field/value (`region:ap-northwest-1`) which will trigger the error in the flyout

<img width="1679" alt="image" src="https://user-images.githubusercontent.com/230104/231422556-b7b3a4be-8ade-4b4c-bab1-a457ab2a0344.png">